api/client.go

```diff
@@ -1097,6 +1097,12 @@ func (c *Client) RawRequest(r *Request) (*Response, error) {
 // a Vault server not configured with this client. This is an advanced operation
 // that generally won't need to be called externally.
 func (c *Client) RawRequestWithContext(ctx context.Context, r *Request) (*Response, error) {
+	ctx, cancelFunc := c.withConfiguredTimeout(ctx)
+	defer cancelFunc()
```

Comment:
Unfortunately, we can't cancel the context here because it will cause the `request.Body` to be invalidated (EOF) before the calling function is done with it. This was the reason for the original hack, which was trying to address #4973 issue. So the function(s) that call `RawRequestWithContext` should be responsible for cancelling the context. I think it would be best to not cancel here and add back the comment explaining the hack.

In my opinion, `RawRequestWithContext` should not have been exported -- this would prevent the issues we are dealing with here.

Reply:
I think we can't unexport `RawRequestWithContext` at the moment anyway, so I assume to have my changes but remove the cancel function call:

```go
func (c *Client) RawRequestWithContext(ctx context.Context, r *Request) (*Response, error) {
	// some meaningful description
	ctx, _ := c.withConfiguredTimeout(ctx)
	return c.rawRequestWithContext(ctx, r)
}
```

IMO it should work the same way as if the context was inside the `rawRequestWithContext` method but just looks cleaner to me. Maybe I missing some part of leakage magic.

Reply:
Done